src/gui_common/charts/line/LineChart.cs

```diff
@@ -0,0 +1,591 @@
+
+            drawArea.DrawTextureRect(hLineTexture, new Rect2(new Vector2(
+                0, ConvertToYCoordinate(value)), RectSize.x, 1), false, new Color(1, 1, 1, 0.3f));
+        }
+    }
+
+    /// <summary>
+    ///   Connect the dataset points with line segments
+    /// </summary>
+    private void DrawLineSegments()
+    {
+        foreach (var data in DataSets)
+        {
+            // Setup the points
+            data.Value.DataPoints.ForEach(point =>
+            {
+                point.Coordinate = ConvertToCoordinate(point.Value);
+
+                if (!point.IsInsideTree())
+                    drawArea.AddChild(point);
+
+                var pointHasZeroValue = point.Value.x <= 0 || point.Value.y <= 0;
+
+                point.Draw = pointHasZeroValue ? false : true;
+
+                // Keep draw if this flag is set to true
+                if (point.DrawAtZeroValue)
+                {
+                    point.Draw = true;
+
+                    // Offset the marker position so it doesn't overlap the axes line
+
+                    if (point.Value.x <= 0)
+                        point.Coordinate += new Vector2(point.Size, 0);
+
+                    if (point.Value.y <= 0)
+                        point.Coordinate -= new Vector2(0, point.Size);
+                }
+            });
+
+            var previousPoint = data.Value.DataPoints.First();
+
+            // Draw the lines
+            foreach (var point in data.Value.DataPoints)
+            {
+                if (data.Value.Draw)
+                {
+                    drawArea.DrawLine(previousPoint.Coordinate, point.Coordinate, data.Value.DataColor,
+                        data.Value.LineWidth, true);
+                }
+
+                previousPoint = point;
+            }
+        }
+    }
+
+    /// <summary>
+    ///   Helper method for converting a single point data value into a coordinate.
+    /// </summary>
+    /// <returns>Position of the given value on the chart</returns>
+    private Vector2 ConvertToCoordinate(Vector2 value)
+    {
+        return new Vector2(ConvertToXCoordinate(value.x), ConvertToYCoordinate(value.y));
+    }
+
+    private float ConvertToXCoordinate(float value)
+    {
+        var lineRectX = drawArea.RectSize.x / XAxisTicks;
+
+        var lineRectWidth = lineRectX * XAxisTicks;
+
+        var dx = MaxValues.x - MinValues.x;
+
+        return (value - MinValues.x) * lineRectWidth / dx;
+    }
+
+    private float ConvertToYCoordinate(float value)
+    {
+        var lineRectY = drawArea.RectSize.y / YAxisTicks;
+
+        var lineRectHeight = lineRectY * (YAxisTicks - 1);
+
+        var dy = MaxValues.y - MinValues.y;
+
+        return lineRectHeight - ((value - MinValues.y) * lineRectHeight / dy) + lineRectY;
+    }
+
+    /// <summary>
+    ///   Returns true if the number of datasets is more than the allowed maximum amount displayed.
+    /// </summary>
+    private bool CheckMaxDataSetShown()
```

Comment:
I think this could be changed into a property, as I've done similar properties elsewhere that check some state in subobjects.